uint8 UNKNOWN=0
uint8 AUTO=1
uint8 MANUAL=2
uint8 mode
